### ISSUE TITLE
BigQuery: detect hidden _PARTITIONTIME column for partitioned tables

### ISSUE THREAD

**Issue by tsdlovell** [ibis-project/ibis#1297]
would be nice to automagically rename it as well

https://cloud.google.com/bigquery/docs/partitioned-tables
> To select the value of _PARTITIONTIME, you must use an alias. For example, the following query selects _PARTITIONTIME by assigning the alias pt to the pseudo column:

**Comment by cpcloud** [comment]
@kszucs can you take a look at this one? you could probably do this by checking the result of `list_partitions()` and then appending an additional field to the bigquery schema and in the client calling

```python
t = super(BigQueryClient, self).table(name)
return t.mutate(PARTITION_TIME=t._PARTITIONTIME).drop(['_PARTITIONTIME'])
```

This would then depend on #1299

**Comment by cpcloud** [comment]
Closed by #1322
